finish_release.py

```diff
@@ -40,6 +44,39 @@ def tag_release(version):
     check_call(['git', 'push', '--follow-tags'])
 
 
+def set_release_date(version):
+    """Sets the release date(s) in RELEASE.rst for any versions missing it"""
+    print("Setting release date...")
+    release_filename = "RELEASE.rst"
+    if not os.path.isfile(release_filename):
+        return
+    date_format = "%Y-%m-%d"
+    check_call(["git", "fetch", "--tags"])
+
+    with open(release_filename) as f:
+        existing_note_lines = [line for line in f.readlines()]
+
+    try:
+        with open(release_filename, "w") as f:
+            for line in existing_note_lines:
+                if line.startswith("Version ") and "Released" not in line:
+                    version_match = re.search(r"[0-9\.]+", line)
+                    if version_match:
+                        version_line = version_match.group(0)
+                        version_date = check_output(
+                            ["git", "log", "-1", "--format=%ai", "v{}".format(version_line)]
+                        ).rstrip()
+                        localtime = datetime.strptime(version_date.decode("utf-8"), "%Y-%m-%d %H:%M:%S %z").\
+                            astimezone(pytz.timezone("America/New_York")).strftime(date_format)
+                        line = "Version {} (Released {})\n".format(version_line, localtime)
+                f.write(line)
+    except:  # pylint:disable=bare-except
+        check_call(["git", "checkout", "HEAD", "--", release_filename])
```

Comment:
We are always working in a temporary directory so we don't need to clean up on exceptions here. Once the exception raises up past `init_working_dir` the directory will automatically get deleted

Reply:
Removing try..except block